web/controllers/machine_controller.ex

```diff
@@ -75,6 +75,18 @@ defmodule CodeStats.MachineController do
       end
   end
 
+  def deactivate(conn, %{"id" => id}) do
+    user = SetSessionUser.get_user_data(conn)
+
+    with %Machine{} = machine <- get_machine_or_404(conn, user, id),
+      changeset                = Machine.update_changeset(machine, %{active: false}),
+      %Machine{} = machine    <- edit_machine_or_flash(conn, changeset) do
+        conn
+        |> put_flash(:success, "Machine #{machine.name} deactivated.")
+        |> redirect(to: machine_path(conn, :list))
+      end
+  end
```

Comment:
Good use of with. The `do` I'd like to be on its own line so that the indentation stays saner but I'll have to check what the official formatter coming in 1.6 says about that.

Reply:
This needs handling for the failure cases (only the happy path is implemented).

Reply:
I thought `edit_machine_or_flash` handles the errors. Isn't that enough?

Reply:
Ah yes indeed, I have been cleverer than myself when I wrote that. Probably not the clearest of codes. Should look into how to rewrite it with https://hexdocs.pm/phoenix/Phoenix.Controller.html#action_fallback/1 later but this will do for now.